src/unit_tests/analysisd/test_decoder_syscollector.c

```diff
@@ -1493,117 +3030,190 @@ void test_syscollector_dbsync_packages_valid_msg_inserted(void **state)
+        expect_any(__wrap_wdbc_query_ex, len);
+        will_return(__wrap_wdbc_query_ex, result);
+        will_return(__wrap_wdbc_query_ex, 0);
+    }
 
     expect_any(__wrap_wdbc_query_ex, *sock);
-    expect_string(__wrap_wdbc_query_ex, query, query);
+    expect_string(__wrap_wdbc_query_ex, query, valid_empty_address_array_queries[i]);
     expect_any(__wrap_wdbc_query_ex, len);
-    will_return(__wrap_wdbc_query_ex, result);
-    will_return(__wrap_wdbc_query_ex, 0);
+    if (parse)
+    {
+        will_return(__wrap_wdbc_query_ex, result_not_ok);
+        will_return(__wrap_wdbc_query_ex, 0);
+    }
+    else
+    {
+        will_return(__wrap_wdbc_query_ex, result);
+        will_return(__wrap_wdbc_query_ex, -1);
+    }
+    expect_string(__wrap__merror, formatted_msg, "Unable to send netinfo message to Wazuh DB.");
 
     int ret = DecodeSyscollector(lf, &sock);
 
-    assert_int_not_equal(ret, 0);
+    assert_int_not_equal(ret, -1);
 }
-void test_syscollector_dbsync_ports_valid_msg_inserted(void **state)
+
+void test_syscollector_netinfo_invalid_address_array_empty_queries (void **state)
 {
-    Eventinfo *lf = *state;
+    size_t count = sizeof(valid_empty_address_array_queries) / sizeof(*valid_empty_address_array_queries);
 
-    const char *query = "agent 001 dbsync ports INSERTED "
-        "{"
-            "\"scan_time\":\"2021/10/29 14:26:24\","
-            "\"protocol\":\"47\","
-            "\"local_ip\":\"48\","
-            "\"local_port\":\"49\","
-            "\"remote_ip\":\"50\","
-            "\"remote_port\":\"51\","
-            "\"tx_queue\":\"52\","
-            "\"rx_queue\":\"53\","
-            "\"inode\":\"54\","
-            "\"state\":\"55\","
-            "\"pid\":\"56\","
-            "\"process\":\"57\","
-            "\"checksum\":\"58\","
-            "\"item_id\":\"59\""
-        "}";
+    for (int i = 0; i < count; i++)
+    {
+        test_syscollector_netinfo_invalid_address_array_empty_query (state, i, false);
+        test_syscollector_netinfo_invalid_address_array_empty_query (state, i, true);
+    }
+}
+
+static const char *valid_empty_address_array_free[] = { "agent 001 netinfo save NULL|NULL|"
+                "NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL",
+            "agent 001 netproto save NULL|NULL|0|NULL|NULL|NULL",
+            "agent 001 netaddr save NULL|NULL|0|0.0.0.0|255.255.255.255|0.0.0.1",
+            "agent 001 netproto save NULL|NULL|1|NULL|"
+                "NULL|NULL",
+            "agent 001 netaddr save NULL|NULL|1|0000:0000:0000:0:0000:0000:0000:0000|"
+                "0000::|0000:0000:0000:0:0000:0000:0000:0001" };
+
+void test_syscollector_netinfo_net_array_data_free (void **state, const int edge, const bool parse)
+{
+    Eventinfo *lf = *state;
     const char *result = "ok ";
+    const char *result_not_ok = "not ok ";
     int sock = 1;
+    int i = 0;
+
+    for (; i < edge; i++)
+    {
+        expect_any(__wrap_wdbc_query_ex, *sock);
```

Comment:
If we don't need to check the value at all, we could use [expect_any_always()](https://api.cmocka.org/group__cmocka__param.html#ga6ad2b0e99309a83d3f3945853ae897b4) outside the loop.

Reply:
I need to repeat a sequence of functions